demos/tests/run_tests.py

```diff
@@ -77,14 +77,15 @@ def temp_dir_as_path():
     with tempfile.TemporaryDirectory() as temp_dir:
         yield Path(temp_dir)
 
+
 def prepare_models(auto_tools_dir, downloader_cache_dir, mo_path, global_temp_dir, demos_to_test):
     model_args = [arg
         for demo in demos_to_test
         for case in demo.test_cases
         for arg in case.options.values()
-        if isinstance(arg, ModelArg)]
+        if isinstance(arg, (ModelArg, ModelFileArg))]
 
-    model_names = {arg.name for arg in model_args}
+    model_names = {arg.name if isinstance(arg, ModelArg) else arg.model_name for arg in model_args}
```

Comment:
IMO, this is getting too hairy. I would suggest to instead move the logic for determining which models are needed into the argument classes themselves. For example, add a base class:

```python
class Arg:
    def required_models(self):
        return []
```

Then make all argument classes inherit it and override where needed:

```python
class ModelArg(Arg):
    # ...
   
    def required_models(self):
        return [self.name, [self.precision]]

class ModelFileArg(Arg):
    # ...

    def required_models(self):
        return [self.model_name, []]
```

Then in this method you can do:

```python
model_names = {}
model_precisions = {}

for demo in demos_to_test:
    for case in demo.test_cases:
        for arg in case.options.values():
            for name, precisions in arg.required_models():
                model_names.add(name)
                model_precisions.update(precisions)

if not model_precisions:
     model_precisions.add('FP32')
```

Reply:
I made some mistakes in the proposed code, should be fixed now.

Reply:
but we still need to check isinstance, because option values can be also strings or None

Reply:
True, but at least there will only be one check.

Reply:
please take a look, I integrated your suggestion